server/src/test/java/org/apache/druid/server/coordinator/duty/CompactSegmentsTest.java

```diff
@@ -2181,7 +2199,7 @@ public void testFindMaxNumTaskSlotsUsedByOneCompactionTaskWhenIsSequentialMode()
       ClientCompactionTaskQueryTuningConfig tuningConfig = Mockito.mock(ClientCompactionTaskQueryTuningConfig.class);
       Mockito.when(tuningConfig.getPartitionsSpec()).thenReturn(Mockito.mock(PartitionsSpec.class));
       Mockito.when(tuningConfig.getMaxNumConcurrentSubTasks()).thenReturn(1);
-      Assert.assertEquals(1, CompactSegments.findMaxNumTaskSlotsUsedByOneCompactionTask(tuningConfig));
+      Assert.assertEquals(1, CompactSegments.findMaxNumTaskSlotsUsedByOneNativeCompactionTask(tuningConfig));
```

Comment:
We should add/modify a test to verify the computation of num task slots used by one MSQ task.

This class also needs to have a couple of tests to verify submission of an MSQ compaction payload. We shouldn't need to have a dependency on the msq extension for it. Best way would be just add engine as a parameter to this test class and run _all_ tests with both engines. Please let me know if you need any assistance in implementing this.

Reply:
Parameterized existing tests and added one where `maxNumTasks` param is added explicitly.